javascript/xhr.py

```diff
@@ -164,7 +164,12 @@ def go():
 
     def send(self, data=None):
         if data is not None:
-            self._request.data = str(data)
+            if str(data) == '[object ArrayBuffer]':
```

Comment:
This will break if someone happens to send the string `"[object ArrayBuffer]"`. You want to check for a string type first:

```python
if isinstance(data, basestring):
    self._request.data = str(data)
elif str(data) == '[object ArrayBuffer]':
    # ...